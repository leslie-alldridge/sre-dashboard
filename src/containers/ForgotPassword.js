import React, { Component } from "react";
import axios from "axios";

import TextField from "@material-ui/core/TextField";

import {
  LinkButtons,
  SubmitButtons,
  registerButton,
  homeButton,
  forgotButton,
  inputStyle,
  HeaderBar
} from "../components";

const title = {
  pageTitle: "Forgot Password Screen"
};

class ForgotPassword extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      showError: false,
      messageFromServer: ""
    };
  }

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  sendEmail = e => {
    e.preventDefault();
    if (this.state.email === "") {
      this.setState({
        showError: false,
        messageFromServer: ""
      });
    } else {
      axios
        .post("/forgotpassword", {
          email: this.state.email
        })
        .then(response => {
          if (response.data === "email not in db") {
            this.setState({
              showError: true,
              messageFromServer: ""
            });
          } else if (response.data === "recovery email sent") {
            this.setState({
              showError: false,
              messageFromServer: "recovery email sent"
            });
          }
        })
        .catch(error => {});
    }
  };

  render() {
    const { email, messageFromServer, showNullError, showError } = this.state;

    return (
      <div>
        <HeaderBar title={title} />
        <form className="profile-form" onSubmit={this.sendEmail}>
          <TextField
            style={inputStyle}
            id="email"
            label="email"
            value={email}
            onChange={this.handleChange("email")}
            placeholder="Email Address"
          />
          <SubmitButtons
            buttonStyle={forgotButton}
            buttonText={"Send Password Reset Email"}
          />
        </form>
        {showNullError && (
          <div>
            <p>The email address cannot be null.</p>
          </div>
        )}
        {showError && (
          <div>
            <p>
              That email address isn't recognized. Please try again or register
              for a new account.
            </p>
            <LinkButtons
              buttonText={`Register`}
              buttonStyle={registerButton}
              link={"/register"}
            />
          </div>
        )}
        {messageFromServer === "recovery email sent" && (
          <div>
            <h3>Password Reset Email Successfully Sent!</h3>
          </div>
        )}
        <LinkButtons
          buttonText={`Go Home`}
          buttonStyle={homeButton}
          link={"/"}
        />
      </div>
    );
  }
}
export default ForgotPassword;
