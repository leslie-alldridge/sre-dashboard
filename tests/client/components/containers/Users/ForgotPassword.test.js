import React from "react";
import { shallow } from "enzyme";
import ForgotPassword from "../../../../../src/containers/Users/ForgotPassword";
describe("MyComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ForgotPassword debug />);

    expect(component).toMatchSnapshot();
  });
});
