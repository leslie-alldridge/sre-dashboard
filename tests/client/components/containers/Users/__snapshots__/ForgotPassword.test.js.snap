// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ForgotPassword
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <HeaderBar
          title={
            Object {
              "pageTitle": "Forgot Password Screen",
            }
          }
        />,
        <form
          className="profile-form"
          onSubmit={[Function]}
        >
          <TextField
            id="email"
            label="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={false}
            select={false}
            style={
              Object {
                "margin": ".5em",
              }
            }
            value=""
          />
          <SubmitButtons
            buttonStyle={
              Object {
                "background": "purple",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Send Password Reset Email"
          />
        </form>,
        undefined,
        false,
        false,
        <LinkButtons
          buttonStyle={
            Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            }
          }
          buttonText="Go Home"
          link="/"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "title": Object {
            "pageTitle": "Forgot Password Screen",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TextField
              id="email"
              label="email"
              onChange={[Function]}
              placeholder="Email Address"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />,
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "purple",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Send Password Reset Email"
            />,
          ],
          "className": "profile-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "id": "email",
              "label": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "required": false,
              "select": false,
              "style": Object {
                "margin": ".5em",
              },
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "buttonStyle": Object {
                "background": "purple",
                "margin": "1em",
                "padding": "1em",
              },
              "buttonText": "Send Password Reset Email",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      undefined,
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttonStyle": Object {
            "background": "mediumpurple",
            "margin": "1em",
            "padding": "1em",
          },
          "buttonText": "Go Home",
          "link": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <HeaderBar
            title={
              Object {
                "pageTitle": "Forgot Password Screen",
              }
            }
          />,
          <form
            className="profile-form"
            onSubmit={[Function]}
          >
            <TextField
              id="email"
              label="email"
              onChange={[Function]}
              placeholder="Email Address"
              required={false}
              select={false}
              style={
                Object {
                  "margin": ".5em",
                }
              }
              value=""
            />
            <SubmitButtons
              buttonStyle={
                Object {
                  "background": "purple",
                  "margin": "1em",
                  "padding": "1em",
                }
              }
              buttonText="Send Password Reset Email"
            />
          </form>,
          undefined,
          false,
          false,
          <LinkButtons
            buttonStyle={
              Object {
                "background": "mediumpurple",
                "margin": "1em",
                "padding": "1em",
              }
            }
            buttonText="Go Home"
            link="/"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "title": Object {
              "pageTitle": "Forgot Password Screen",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TextField
                id="email"
                label="email"
                onChange={[Function]}
                placeholder="Email Address"
                required={false}
                select={false}
                style={
                  Object {
                    "margin": ".5em",
                  }
                }
                value=""
              />,
              <SubmitButtons
                buttonStyle={
                  Object {
                    "background": "purple",
                    "margin": "1em",
                    "padding": "1em",
                  }
                }
                buttonText="Send Password Reset Email"
              />,
            ],
            "className": "profile-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "id": "email",
                "label": "email",
                "onChange": [Function],
                "placeholder": "Email Address",
                "required": false,
                "select": false,
                "style": Object {
                  "margin": ".5em",
                },
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "buttonStyle": Object {
                  "background": "purple",
                  "margin": "1em",
                  "padding": "1em",
                },
                "buttonText": "Send Password Reset Email",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "form",
        },
        undefined,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttonStyle": Object {
              "background": "mediumpurple",
              "margin": "1em",
              "padding": "1em",
            },
            "buttonText": "Go Home",
            "link": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
